dotCMS/src/main/java/com/dotmarketing/portlets/variant/business/VariantAPIImpl.java

```diff
@@ -0,0 +1,89 @@
+package com.dotmarketing.portlets.variant.business;
+
+import com.dotcms.business.WrapInTransaction;
+import com.dotcms.rest.validation.Preconditions;
+import com.dotmarketing.business.FactoryLocator;
+import com.dotmarketing.exception.DoesNotExistException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.variant.model.Variant;
+import java.util.Optional;
+
+/**
+ * API of {@link Variant}
+ */
+public class VariantAPIImpl implements VariantAPI {
+    private VariantFactory variantFactory;
+
+    public VariantAPIImpl(){
+        variantFactory = FactoryLocator.getVariantFactory();
+    }
+
+    /**
+     * Implementation for {@link VariantAPI#save(Variant)}
+     * @param variant
+     *
+     * @return
+     * @throws DotDataException
+     */
+    @Override
+    @WrapInTransaction
+    public Variant save(final Variant variant) throws DotDataException {
+
+        Preconditions.checkNotNull(variant.getName(), "Variant name should not be null");
+
+        if (variant.isArchived()) {
+            throw new IllegalArgumentException("Variant can not be created as archive");
+        }
+
+        return variantFactory.save(variant);
+    }
+
+    /**
+     * Implementation for {@link VariantAPI#update(Variant)}
+     * @param variant
+     *
+     * @throws DotDataException
+     */
+    @Override
+    public void update(final Variant variant) throws DotDataException {
+        Preconditions.checkNotNull(variant.getName(), "Variant name should not be null");
```

Comment:
u should check the variant to update exists, also wrap in tran

Reply:
done https://github.com/dotCMS/core/pull/22839/commits/19c1fec4f3218ef29e0a009a1bf52c3402de51c7#diff-a2746c1b6a4b623f65d595e778def0a3d7ce80f05cf45d412a4472b1e6b1bf92R57